README.md

```diff
@@ -1,2 +1,101 @@
 # ipni
-Create signed advertisments for IPNI
+
+> Create signed advertisements for the [InterPlanetary Network Indexer](https://github.com/ipni/specs/blob/main/IPNI.md)
+
+Supports single and [extended providers](https://github.com/ipni/specs/blob/main/IPNI.md#extendedprovider).
+
+Derived from reference implentation in https://github.com/ipni/go-libipni/blob/main/ingest/schema/envelope.go
+
+See the [IPLD Schema](./schema.ipldsch) for the encoded Advertisement shape. The encoding logic in this lib is validated against that schema.
+
+## Single provider
+
+Encode an signed advertisement for a new batch of entries available from a single provider
+
+```js
+import test from 'ava'
+import { CID } from 'multiformats/cid'
+import { createEd25519PeerId } from '@libp2p/peer-id-factory'
+import { Provider, Advertisement } from '@web3-storage/ipni'
+
+// Link to the latest batch of multihashes
+const entries = CID.parse('bafybeiczsscdsbs7ffqz55asqdf3smv6klcw3gofszvwlyarci47bgf354') 
```

Comment:
It might be nice for completeness to have an `Entries` class that just spits out an "encoded" object like `{ Entries: Link[] }`.

```suggestion
import { Provider, Advertisement, Entries } from '@web3-storage/ipni'
import { Block } from 'multiformats/block'
import { sha256 as hasher } from 'multiformats/hashes/sha2'
import * as codec from '@ipld/dag-json'

const ents = new Entries([
  CID.parse('bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy'),
  // ...
])
const entsBlock = Block.encode({ value: ents.encode(), codec, hasher })

// Link to the latest batch of multihashes
const entries = entsBlock.cid
```

Reply:
Raised issue https://github.com/web3-storage/ipni/issues/2